components/browser/session/src/main/java/mozilla/components/browser/session/SessionManager.kt

```diff
@@ -166,6 +167,33 @@ class SessionManager(
         ))
     }
 
+    /**
+     * Restores the given list of [RecoverableTab].
+     */
+    fun restore(tabs: List<RecoverableTab>) {
+        // As a workaround we squint here and pretend this is a Snapshot..
+
+        val items = tabs.map {
+            Snapshot.Item(
+                session = Session(
+                    id = it.id,
+                    initialUrl = it.url,
+                    contextId = it.contextId
+                ).apply {
+                    title = it.title
+                    parentId = it.parentId
+                },
+                engineSessionState = it.state,
+                readerState = it.readerState,
+                lastAccess = it.lastAccess
+            )
+        }
+
+        val snapshot = Snapshot(items, NO_SELECTION)
+
+        restore(snapshot, updateSelection = false)
+    }
+
```

Comment:
Should we have a test on `SessionManagerTest` for this functionality?

Reply:
Added another test. But this is also covered through the middleware test.